---
title: "CV1.2 Data exploration and normalization subsets"
author: "C.deVriend"
date: "10-7-2022"
output: html_document
toc: true
toc_float: true
number_sections: true
theme: readable
code_folding: hide
---

```{css, echo=FALSE}
pre {
  max-height: 300px;
  overflow-y: auto;
}
pre[class] {
  max-height: 100px;
}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Raw expression data
The bulk PAS-seq dataset described by Kim D. S. et al., (2020) is based upon induced keratinocyte differentiation by addition of calcium. For a total of six days, cells were harvested every twelve hours creating a total of thirteen time points. This allows for comprehensive data that can be analyzed extensively in order to identify transcriptional activity during the variety of differentiation time points. 

Note that the the raw counts data (57820 genes), which was used in this analysis, only contains ten time points in contrast to the thirteen time points mentioned by Kim D. S. et al., (2020).
```{r}
library(here)
set_here(path = "C:/Users/cheye/OneDrive/Bureaublad/Radboud/Master 21-22/Internship/Data and analysis/CV1 Data exploration and normalization")
# Check if i am working in correct directory
here() 

# load the raw counts data and give it a nice name
raw_counts <- read.delim("../data/Raw count matrix/ggr.rna.counts.mat.txt.gz", row.names=1)
```
```{r}
# remove counts below 1
raw_counts<-raw_counts[rowSums(raw_counts)>1,]
``` 
```{r}
# Annotation
library(biomaRt)
mart= useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
genes= getBM(attributes = c("ensembl_gene_id", "hgnc_symbol"), values = rownames(raw_counts), mart= mart)
matched= match(rownames(raw_counts), genes$ensembl_gene_id)
raw_counts$ID=genes$hgnc_symbol[matched]
# Remove duplicate rowname 'POLR2J4'
raw_counts<-raw_counts[!(raw_counts$ID=="POLR2J4"),]

# change empty spaces with NA then remove NA's 
raw_counts[raw_counts == ""] <- NA
# Remove NA's # we go from 19144 rows to 18567 rows
raw_counts<- na.omit(raw_counts)

# load library
library(tibble)

# Change rownames
raw_counts<-rownames_to_column(raw_counts)
raw_counts<-column_to_rownames(raw_counts, var = "ID")

# Remove additional column with ensembl ID
raw_counts<-dplyr::select(raw_counts, -c(rowname))


# left with 18567 genes
```

## Subsetting count data per time point
In order to compare a relevant combination of keratinocyte differentiation time points to each other the data columns must be subsetted. In order to perform differential expression analysis, a total of 10 subsets will be made. The reason we aren't working with the mean data per timepoint, is because DESeq prefers to have duplicates. Without duplicates we would need to force a fit by e.g.~ 0 + condition, which can make the results and specifcally the logfold changes unreliable.

*Subsets:*

1. Comparison day 0 and day 0.5

2. Comparison day 0.5 and day 1

3. Comparison day 1 and day 1.5

4. Comparison day 1.5 and day 2

5. Comparison day 2 and day 2.5

6. Comparison day 2.5 and day 3

7. Comparison day 3 and day 4.5

8. Comparison day 4.5 and day 5

9. Comparison day 5 and day 6

10. Comparison day 6 and day 0
 
```{r}

# Create dataset comparison names (10 total)
Datasets_comparison<- c("Comparison day 0 and day 0.5", "Comparison day 0.5 and day 1", "Comparison day 1 and day 1.5", "Comparison day 1.5 and day 2", "Comparison day 2 and day 2.5", "Comparison day 2.5 and day 3", "Comparison day 3 and day 4.5", "Comparison day 4.5 and day 5", "Comparison day 5 and day 6", "Comparison day 6 and day 0")

# Create list that contains all the subsets
Comparison_.list <- list()

for (i in seq_along(Datasets_comparison)) {
  data.comparison <- raw_counts
  Comparison_.list[[Datasets_comparison[i]]] <- data.comparison 
}

# Alter df's in comparison.list so it contains only the columns of the comparison we need
## Comparison day 0 and day 0.5 ##
# Remove columns 5:20
Comparison_.list$`Comparison day 0 and day 0.5`<-Comparison_.list$`Comparison day 0 and day 0.5`[-c(5:20)]

## Comparison day 0.5 and day 1 ##
# Remove columns 1:2 and 7:20
Comparison_.list$`Comparison day 0.5 and day 1`<-Comparison_.list$`Comparison day 0.5 and day 1`[-c(1:2, 7:20)]

## Comparison day 1 and day 1.5 ## 
# Remove columns 1:4 and 9:20
Comparison_.list$`Comparison day 1 and day 1.5`<-Comparison_.list$`Comparison day 1 and day 1.5`[-c(1:4, 9:20)]

## Comparison day 1.5 and and day 2 ##
# Remove columns 1:6 and 11:20
Comparison_.list$`Comparison day 1.5 and day 2`<-Comparison_.list$`Comparison day 1.5 and day 2`[-c(1:6, 11:20)]

## Comparison day 2 and day 2.5 ##
# Remove columns 1:8 and 13:20
Comparison_.list$`Comparison day 2 and day 2.5`<-Comparison_.list$`Comparison day 2 and day 2.5`[-c(1:8, 13:20)]

## Comparison day 2.5 and day 3 ##
# Remove columns 1:10 15:20
Comparison_.list$`Comparison day 2.5 and day 3`<-Comparison_.list$`Comparison day 2.5 and day 3`[-c(1:10, 15:20)]

## Comparison day 3 and day 4.5 ##
# Remove columns 1:12 and 17:20
Comparison_.list$`Comparison day 3 and day 4.5`<-Comparison_.list$`Comparison day 3 and day 4.5`[-c(1:12, 17:20)]

## Comparison day 4.5 and day 5 ##
# Remove columns 1:14 and 19:20
Comparison_.list$`Comparison day 4.5 and day 5`<-Comparison_.list$`Comparison day 4.5 and day 5`[-c(1:14, 19:20)]

## Comparison day 5 and day 6 ##
# Remove columns 1:16
Comparison_.list$`Comparison day 5 and day 6`<-Comparison_.list$`Comparison day 5 and day 6`[-c(1:16)]

## Comparison day 6 and day 0 ##
# Remove columns 3:18
Comparison_.list$`Comparison day 6 and day 0`<-Comparison_.list$`Comparison day 6 and day 0`[-c(3:18)]
```

## Data normalization and filtering
In RNA-seq data the expected variance grows with the mean, which means the data is heteroskedastic. To be specific, for RNA-seq data the variance actually grows faster than the mean as it has a negative binomial distribution. Therefore, results from e.g. a principal component analysis (PCA) will be driven by genes with the highest counts because those are the ones which naturally tend to show the largest absolute differences between samples. 

As a solution, the DESeq2  package offers two transformations for count data that stabilize the variance across the mean. The variance stabilizing transformation (VST) for negative binomial data with a dispersion-mean trent, implemented in the vst.function and the regularized-logarithm transformation or rlog.

```{r}
#FIRST CREATE A DESeq OBJECT FROM EACH SUBSET 
#Load DESeq2 library
library(DESeq2)

# The experimental design is specified at the beginning of the analysis, as it will inform many of the DESeq2 functions how to treat the samples in the analysis. The design formula tells which columns in the sample info table specify the experimental design and how these factors should be used in the analysis.
# The simplest design formula for differential expression would be ~condition, where condition is a column in the sample info that specifies which of two (or more) groups the samples belong to. In our case we would use ~ bulk_set + Timepoint_days.

## Total dataset ##
# Create sample info columns
Names<-c('Day_0', 'Day_0_dup', 'Day_0.5', 'Day_0.5_dup', 'Day_1', 'Day_1_dup', 'Day_1.5', 'Day_1.5_dup', 'Day_2', 'Day_2_dup', 'Day_2.5',  'Day_2.5_dup', 'Day_3', 'Day_3_dup', 'Day_4.5', 'Day_4.5_dup', 'Day_5', 'Day_5_dup', 'Day_6', 'Day_6_dup')
Timepoint<-(c(0,0,0.5,0.5,1,1,1.5,1.5,2,2,2.5,2.5,3,3,4.5,4.5,5,5,6,6))
bulk<-c('original','duplicate', 'original','duplicate','original','duplicate', 'original','duplicate','original','duplicate', 'original','duplicate','original','duplicate', 'original','duplicate','original','duplicate', 'original','duplicate')
info<-as.data.frame(Names)
info$Timepoint<-(Timepoint)
info$bulk<-(bulk)
dds_total<-DESeqDataSetFromMatrix(countData = raw_counts, colData = info, design = ~ bulk + Timepoint)
# Remove low counts
dds_total<-dds_total[rowSums(counts(dds_total)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds_total=estimateSizeFactors(dds_total)
dds_total=estimateDispersions(dds_total)
gene_total_set=getVarianceStabilizedData(dds_total)


## Comparison day 0 and day 0.5 ##
# Create sample information columns
Names_0_0.5<-c('Day_0', 'Day_0_dup', 'Day_0.5', 'Day_0.5_dup') 
Timepoint_0_0.5<-c(0,0,0.5,0.5)
bulk_0_0.5<-c('original','duplicate', 'original','duplicate')
info_0_0.5<-as.data.frame(Names_0_0.5)
info_0_0.5$Timepoint_0_0.5<-(Timepoint_0_0.5)
info_0_0.5$bulk_0_0.5<-(bulk_0_0.5)
dds0_0.5 <-DESeqDataSetFromMatrix(countData = Comparison_.list$`Comparison day 0 and day 0.5`, colData = info_0_0.5, design = ~ bulk_0_0.5 + Timepoint_0_0.5)
# Remove low counts
dds0_0.5<-dds0_0.5[rowSums(counts(dds0_0.5)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds0_0.5=estimateSizeFactors(dds0_0.5)
dds0_0.5=estimateDispersions(dds0_0.5)
gene_expr_0_0.5=getVarianceStabilizedData(dds0_0.5)

## Comparison day 0.5 and day 1 ##
Names_0.5_1<-c('Day_0.5', 'Day_0.5_dup', 'Day_1', 'Day_1_dup') 
Timepoint_0.5_1<-c(0.5,0.5,1,1)
bulk_0.5_1<-c('original','duplicate', 'original','duplicate')
info_0.5_1<-data.frame(Names_0.5_1, bulk_0.5_1, Timepoint_0.5_1)
dds0.5_1 <-DESeqDataSetFromMatrix(countData = Comparison_.list$`Comparison day 0.5 and day 1`, colData = info_0.5_1, design = ~ bulk_0.5_1 + Timepoint_0.5_1)
# Remove low counts
dds0.5_1<-dds0.5_1[rowSums(counts(dds0.5_1)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds0.5_1=estimateSizeFactors(dds0.5_1)
dds0.5_1=estimateDispersions(dds0.5_1)
gene_expr_0.5_1=getVarianceStabilizedData(dds0.5_1)

## Comparison day 1 and day 1.5 ##
Names_1_1.5<-c('Day_1', 'Day_1_dup', 'Day_1.5', 'Day_1.5_dup') 
Timepoint_1_1.5<-c(1,1,1.5,1.5)
bulk_1_1.5<-c('original','duplicate', 'original','duplicate')
info_1_1.5<-data.frame(Names_1_1.5, bulk_1_1.5, Timepoint_1_1.5)
dds1_1.5 <-DESeqDataSetFromMatrix(countData = Comparison_.list$`Comparison day 1 and day 1.5`, colData = info_1_1.5, design = ~ bulk_1_1.5 + Timepoint_1_1.5)
# Remove low counts
dds1_1.5<-dds1_1.5[rowSums(counts(dds1_1.5)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds1_1.5=estimateSizeFactors(dds1_1.5)
dds1_1.5=estimateDispersions(dds1_1.5)
gene_expr1_1.5=getVarianceStabilizedData(dds1_1.5)

## Comparison day 1.5 and day 2 ##
Names_1.5_2<-c('Day_1.5', 'Day_1.5_dup', 'Day_2', 'Day_2_dup') 
Timepoint_1.5_2<-c(1.5,1.5,2,2)
bulk_1.5_2<-c('original','duplicate', 'original','duplicate')
info_1.5_2<-data.frame(Names_1.5_2, bulk_1.5_2, Timepoint_1.5_2)
dds1.5_2 <-DESeqDataSetFromMatrix(countData = Comparison_.list$`Comparison day 1.5 and day 2`, colData = info_1.5_2, design = ~ bulk_1.5_2 + Timepoint_1.5_2)
# Remove low counts
dds1.5_2<-dds1.5_2[rowSums(counts(dds1.5_2)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds1.5_2=estimateSizeFactors(dds1.5_2)
dds1.5_2=estimateDispersions(dds1.5_2)
gene_expr_1.5_2=getVarianceStabilizedData(dds1.5_2)

## Comparison day 2 and day 2.5 ##
Names_2_2.5<-c('Day_2', 'Day_2_dup', 'Day_2.5', 'Day_2.5_dup') 
Timepoint_2_2.5<-c(2,2,2.5,2.5)
bulk_2_2.5<-c('original','duplicate', 'original','duplicate')
info_2_2.5<-data.frame(Names_2_2.5, bulk_2_2.5, Timepoint_2_2.5)
dds2_2.5 <-DESeqDataSetFromMatrix(countData = Comparison_.list$`Comparison day 2 and day 2.5`, colData = info_2_2.5, design = ~ bulk_2_2.5 + Timepoint_2_2.5)
# Remove low counts
dds2_2.5<-dds2_2.5[rowSums(counts(dds2_2.5)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds2_2.5=estimateSizeFactors(dds2_2.5)
dds2_2.5=estimateDispersions(dds2_2.5)
gene_expr_2_2.5=getVarianceStabilizedData(dds2_2.5)

## Comparison day 2.5 and day 3 ##
Names_2.5_3<-c('Day_2.5', 'Day_2.5_dup', 'Day_3', 'Day_3_dup') 
Timepoint_2.5_3<-c(2.5,2.5,3,3)
bulk_2.5_3<-c('original','duplicate', 'original','duplicate')
info_2.5_3<-data.frame(Names_2.5_3, bulk_2.5_3, Timepoint_2.5_3)
dds2.5_3 <-DESeqDataSetFromMatrix(countData = Comparison_.list$`Comparison day 2.5 and day 3`, colData = info_2.5_3, design = ~ bulk_2.5_3 + Timepoint_2.5_3)
# Remove low counts
dds2.5_3<-dds2.5_3[rowSums(counts(dds2.5_3)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds2.5_3=estimateSizeFactors(dds2.5_3)
dds2.5_3=estimateDispersions(dds2.5_3)
gene_expr_2.5_3=getVarianceStabilizedData(dds2.5_3)

## Comparison day 3 and day 4.5 ##
Names_3_4.5<-c('Day_3', 'Day_3_dup', 'Day_4.5', 'Day_4.5_dup') 
Timepoint_3_4.5<-c(3,3,4.5,4.5)
bulk_3_4.5<-c('original','duplicate', 'original','duplicate')
info_3_4.5<-data.frame(Names_3_4.5, bulk_3_4.5, Timepoint_3_4.5)
dds3_4.5 <-DESeqDataSetFromMatrix(countData = Comparison_.list$`Comparison day 3 and day 4.5`, colData = info_3_4.5, design = ~ bulk_3_4.5 + Timepoint_3_4.5)
# Remove low counts
dds3_4.5<-dds3_4.5[rowSums(counts(dds3_4.5)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds3_4.5=estimateSizeFactors(dds3_4.5)
dds3_4.5=estimateDispersions(dds3_4.5)
gene_expr_3_4.5=getVarianceStabilizedData(dds3_4.5)

## Comparison day 4.5 and day 5 ##
Names_4.5_5<-c('Day_4.5', 'Day_4.5_dup', 'Day_5', 'Day_5_dup') 
Timepoint_4.5_5<-c(4.5,4.5,5,5)
bulk_4.5_5<-c('original','duplicate', 'original','duplicate')
info_4.5_5<-data.frame(Names_4.5_5, bulk_4.5_5, Timepoint_4.5_5)
dds4.5_5<-DESeqDataSetFromMatrix(countData = Comparison_.list$`Comparison day 4.5 and day 5`, colData = info_4.5_5, design = ~ bulk_4.5_5 + Timepoint_4.5_5)
# Remove low counts
dds4.5_5<-dds4.5_5[rowSums(counts(dds4.5_5)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds4.5_5=estimateSizeFactors(dds4.5_5)
dds4.5_5=estimateDispersions(dds4.5_5)
gene_expr_4.5_5=getVarianceStabilizedData(dds4.5_5)

## Comparison day 5 and day 6 ##
Names_5_6<-c('Day_5', 'Day_5_dup', 'Day_6', 'Day_6_dup') 
Timepoint_5_6<-c(5,5,6,6)
bulk_5_6<-c('original','duplicate', 'original','duplicate')
info_5_6<-data.frame(Names_5_6, bulk_5_6, Timepoint_5_6)
dds5_6 <-DESeqDataSetFromMatrix(countData = Comparison_.list$`Comparison day 5 and day 6`, colData = info_5_6, design = ~ bulk_5_6 + Timepoint_5_6)
# Remove low counts
dds5_6<-dds5_6[rowSums(counts(dds5_6)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds5_6=estimateSizeFactors(dds5_6)
dds5_6=estimateDispersions(dds5_6)
gene_expr_5_6=getVarianceStabilizedData(dds5_6)

## Comparison day 6 and day 0 ##
Names_6_0<-c('Day_0', 'Day_0_dup', 'Day_6', 'Day_6_dup') 
Timepoint_6_0.5<-c(0,0,6,6)
bulk_6_0<-c('original','duplicate', 'original','duplicate')
info_6_0<-data.frame(Names_6_0, bulk_6_0, Timepoint_6_0.5)
dds6_0 <-DESeqDataSetFromMatrix(countData = Comparison_.list$`Comparison day 6 and day 0`, colData = info_6_0, design = ~ bulk_6_0 + Timepoint_6_0.5)
# Remove low counts
dds6_0<-dds6_0[rowSums(counts(dds6_0)) >9,]
# Create normalized object for further analyses e.g. dorothea
dds6_0=estimateSizeFactors(dds6_0)
dds6_0=estimateDispersions(dds6_0)
gene_expr_6_0=getVarianceStabilizedData(dds6_0)

# Create a list that keeps all the deseq objects
Deseq_objects.list<-list(dds0_0.5,dds0.5_1,dds1_1.5,dds1.5_2,dds2_2.5,dds2.5_3,dds3_4.5,dds4.5_5,dds5_6,dds6_0)
# Change list names
names(Deseq_objects.list)[1]<-"dds_0_0.5"
names(Deseq_objects.list)[2]<-"dds_0.5_1"
names(Deseq_objects.list)[3]<-"dds_1_1.5"
names(Deseq_objects.list)[4]<-"dds_1.5_2"
names(Deseq_objects.list)[5]<-"dds_2_2.5"
names(Deseq_objects.list)[6]<-"dds_2.5_3"
names(Deseq_objects.list)[7]<-"dds_3_4.5"
names(Deseq_objects.list)[8]<-"dds_4.5_5"
names(Deseq_objects.list)[9]<-"dds_5_6"
names(Deseq_objects.list)[10]<-"dds_6_0"

# Export gene_expr df's to excel for dorothea
library(writexl)

gene_expr_0_0.5<-as.data.frame(gene_expr_0_0.5)

write_xlsx(gene_expr_0_0.5, "gene_expr_0_0.5.xlsx")
write_xlsx(as.data.frame(gene_expr_0.5_1), "gene_expr_0.5_1.xlsx")
write_xlsx(as.data.frame(gene_expr1_1.5), "gene_expr_1_1.5.xlsx")
write_xlsx(as.data.frame(gene_expr_1.5_2), "gene_expr_1.5_2.xlsx")
write_xlsx(as.data.frame(gene_expr_2_2.5), "gene_expr_2_2.5.xlsx")
write_xlsx(as.data.frame(gene_expr_2.5_3), "gene_expr_2.5_3.xlsx")
write_xlsx(as.data.frame(gene_expr_3_4.5), "gene_expr_3_4.5.xlsx")
write_xlsx(as.data.frame(gene_expr_4.5_5), "gene_expr_4.5_5.xlsx")
write_xlsx(as.data.frame(gene_expr_5_6), "gene_expr_5_6.xlsx")
write_xlsx(as.data.frame(gene_expr_6_0), "gene_expr_6_0.xlsx")
write_xlsx(as.data.frame(gene_total_set), "gene_expr_total_set.xlsx")
```

There are 57820 genes in total in this set. Note that genes with counts below 10 were excluded from each separate subset. 

*Subsets:*
  
  1. Comparison day 0 and day 0.5 --> 15529 genes

2. Comparison day 0.5 and day 1 --> 15296 genes

3. Comparison day 1 and day 1.5 --> 15679 genes

4. Comparison day 1.5 and day 2 --> 15479 genes 

5. Comparison day 2 and day 2.5 --> 15481 genes

6. Comparison day 2.5 and day 3 --> 15630 genes

7. Comparison day 3 and day 4.5 --> 15863 genes

8. Comparison day 4.5 and day 5 --> 15810 genes

9. Comparison day 5 and day 6   --> 15777 genes

10. Comparison day 6 and day 0  --> 16166 genes


```{r}
# Remove all counts below 10 and normalize data
# In RNA-seq experiments events where info is not complex or rich enough are called dropouts. Gene dropouts are genes that show low or moderate expression level in one sample but are not detected in any other sample. A sample dropout is when a sample shows any expression in an abnormally low amount of genes compared to the rest of the samples
# Given the nature of the bulk RNA-seq data, sample dropout will not be an issue. However, let's have a look at how many genes we are considering in our matrix and how many of those have 1 count or less across all the samples.
# Remember that genes are rows in the genes by samples (counts) matrix. we can check how many genes are in total by counting rows

# Remember that genes are rows in the genes by samples (counts) matrix. we can check how many genes are in total by counting rows
nrow(counts(dds1_1.5)) # 57820 genes, as expected # doesn't matter which subset you take here all are just as long
# check how many show 10 or less counts; this is different for each subset; you don't have to do this in your own code but just to illustrate how it is variable between sets
sum(rowSums(counts(dds1_1.5))<10) #  show less than 10 counts; this is very low lets exclude these

# --------------------------------------------------------
# Keep only genes with counts above 10
#dds_0_0.5
Deseq_objects.list$dds_0_0.5<-Deseq_objects.list$dds_0_0.5[rowSums(counts(Deseq_objects.list$dds_0_0.5)) >10,]
#the condition will again return a vector of TRUE/FALSE for each gene, which we used to decide what genes to keep
nrow(Deseq_objects.list$dds_0_0.5) #  15529 genes kept

# dds_0.5_1
Deseq_objects.list$dds_0.5_1<-Deseq_objects.list$dds_0.5_1[rowSums(counts(Deseq_objects.list$dds_0.5_1)) >10,]
nrow(Deseq_objects.list$dds_0.5_1) 

# dds_1_1.5
Deseq_objects.list$dds_1_1.5<-Deseq_objects.list$dds_1_1.5[rowSums(counts(Deseq_objects.list$dds_1_1.5)) >10,]
nrow(Deseq_objects.list$dds_1_1.5)

# dds_1.5_2
Deseq_objects.list$dds_1.5_2<-Deseq_objects.list$dds_1.5_2[rowSums(counts(Deseq_objects.list$dds_1.5_2)) >10,]
nrow(Deseq_objects.list$dds_1.5_2)

# dds_2_2.5
#Deseq_objects.list$dds_2_2.5<-Deseq_objects.list$dds_2_2.5[rowSums(counts(Deseq_objects.list$dds_2_2.5)) >10,]
#nrow(Deseq_objects.list$dds_2_2.5)

# dds_2.5_3
Deseq_objects.list$dds_2.5_3<-Deseq_objects.list$dds_2.5_3[rowSums(counts(Deseq_objects.list$dds_2.5_3)) >10,]
nrow(Deseq_objects.list$dds_2.5_3)

# dds_3_4.5
#Deseq_objects.list$dds_3_4.5<-Deseq_objects.list$dds_3_4.5[rowSums(counts(Deseq_objects.list$dds_3_4.5)) >10,]
#nrow(Deseq_objects.list$dds_3_4.5)

# dds_4.5_5
Deseq_objects.list$dds_4.5_5<-Deseq_objects.list$dds_4.5_5[rowSums(counts(Deseq_objects.list$dds_4.5_5)) >10,]
nrow(Deseq_objects.list$dds_4.5_5)

# dds_5_6
Deseq_objects.list$dds_5_6<-Deseq_objects.list$dds_5_6[rowSums(counts(Deseq_objects.list$dds_5_6)) >10,]
nrow(Deseq_objects.list$dds_5_6)

# dds_6_0
Deseq_objects.list$dds_6_0<-Deseq_objects.list$dds_6_0[rowSums(counts(Deseq_objects.list$dds_6_0)) >10,]
nrow(Deseq_objects.list$dds_6_0)

```
```{r}
# Variance stabilizing transformation -------------------------------------
# In RNA-seq data the expected variance grows with the mean (Data is heteroskedastic) hence, PCA results will be driven by genes with the highest counts because they're the ones which tend to naturally show the largest absolute differences between samples. 
# Notice that for RNA-seq data, the variance actually grows faster than the mean (it has a negative bionomial distribution).
# As a solution, DESeq2 offers two transformations for count data that stabilize the variance across the mean: the variance stabilizing transformation (VST) for negative binomial data with a dispersion-mean trent, implemented in the vst function and the regularized-logarithm transformation or rlog.
# Transform data for visualization purposes and PROGENy differential pathway analysis.
# the option blind = FALSE tells the function not to neglect the design; as time point might affect the dispersion-mean as well 

vsd_0_0.5<-vst(Deseq_objects.list$dds_0_0.5, blind = FALSE)
vsd_0.5_1<-vst(Deseq_objects.list$dds_0.5_1, blind = FALSE)
vsd_1_1.5<-vst(Deseq_objects.list$dds_1_1.5, blind = FALSE)
vsd_1.5_2<-vst(Deseq_objects.list$dds_1.5_2, blind = FALSE)
vsd_2_2.5<-vst(Deseq_objects.list$dds_2_2.5, blind = FALSE)
vsd_2.5_3<-vst(Deseq_objects.list$dds_2.5_3, blind = FALSE)
vsd_3_4.5<-vst(Deseq_objects.list$dds_3_4.5, blind = FALSE)
vsd_4.5_5<-vst(Deseq_objects.list$dds_4.5_5, blind = FALSE)
vsd_5_6<-vst(Deseq_objects.list$dds_5_6, blind = FALSE)
vsd_6_0<-vst(Deseq_objects.list$dds_6_0, blind = FALSE)

# Create list that contains normalized subsets
Deseq_normalized_objects.list<-list(vsd_0_0.5   ,vsd_0.5_1,vsd_1_1.5,vsd_1.5_2,vsd_2_2.5,vsd_2.5_3,vsd_3_4.5,vsd_4.5_5,vsd_5_6,vsd_6_0)
# Change list names
names(Deseq_normalized_objects.list)[1]<-"vsd_0_0.5"
names(Deseq_normalized_objects.list)[2]<-"vsd_0.5_1"
names(Deseq_normalized_objects.list)[3]<-"vsd_1_1.5"
names(Deseq_normalized_objects.list)[4]<-"vsd_1.5_2"
names(Deseq_normalized_objects.list)[5]<-"vsd_2_2.5"
names(Deseq_normalized_objects.list)[6]<-"vsd_2.5_3"
names(Deseq_normalized_objects.list)[7]<-"vsd_3_4.5"
names(Deseq_normalized_objects.list)[8]<-"vsd_4.5_5"
names(Deseq_normalized_objects.list)[9]<-"vsd_5_6"
names(Deseq_normalized_objects.list)[10]<-"vsd_6_0"
```

## Differential expression analysis ##
Now the data has been formatted accordingly, the differential expression analyses can be performed. The DEseq2 package has a specific function for this: DEseq(dataset).

## Differential expression analysis - Comparison timepoint 0 and 0.5 ##
```{r}
# I have already specified an experimental design when I create the DESeqDataset, now run the differential expression pipeline on the raw counts with a single call to the function DESeq;
# WE DO NOT USE THE VST DATA FOR THIS; DESEQ PERFORMS NORMALIZATION AS WELL
diffex_0_0.5<-DESeq(dds0_0.5)
# This function prints out a message for the various steps it performs. These are:
# 1. The estimation of size factors, controlling for differences in the counts due to varying sequencing depth of the samples
# 2. The estimation of dispersion values. The dispersion parameter captures how much the counts for the samples will vary around an expected value. Note that the expected value takes into consideration the sequencning depth and differences that can be attributed to variables in the design formula.
# 3. Fitting a final generalized linear model using the size factors ad dispersion values estimated above, which gives estimates of the log fold changes.

# Building the results table
# Calling results without any arguments will extract the estimated log2 fold changes and p values for the last variable of the design formula. The ccomparison is printed at the top of the output: timepoint vs duplicate
res0_0.5 <- results(diffex_0_0.5)
res0_0.5
# first column baseMean is just the average of the normalized count values, divided by the size facotrs, taken over all samples in the DESeqDataSet. the remaining four columns refer to a specifc contrast, namely the comparison of the duplicate status (original or duplicate) for the factor variable
# The column log2foldchange is the effect size estimate. it tells us how much the genes expression seems to have changed between orignal and duplicate.The value is reported on the logarithmic scale to base 2 e.g. a change of gene expression of a factor 8 would give a value 3.
# the column lfcSE displays the standard error estimate for the log2 fold change estimate, which tells me about the uncertainty that's associated with that fold change.
# The purpose of a test for differential expression is to test whether the data provides sufficient evidence to conclude that this value (the difference in expression) is really different from zero. DESeq2 performs for each gene a hypothesis test ot see whether the evidence is sufficient to decide agianst H0. the result of this test can be found in the column p-value.
# Finally the column padj reports the p-value but corrected for multiple testing. By default, Benjamine & Hockberg (FDR) is used for this.

# summary results
summary(res0_0.5)
# Look at results when p-value is 0.05
res0_0.5_p0.05<- results(diffex_0_0.5, alpha = 0.05)
summary(res0_0.5_p0.05)
# Look at results when p-value is 0.01
res0_0.5_p0.01<- results(diffex_0_0.5, alpha = 0.01)
summary(res0_0.5_p0.01)

# Determine expression labels; up down 
res0_0.5_p0.05$expression = ifelse(res0_0.5_p0.05$padj < 0.05 & abs(res0_0.5_p0.05$log2FoldChange) >= 1, 
                     ifelse(res0_0.5_p0.05$log2FoldChange> 1 ,'Up','Down'),
                     'Stable')
res0_0.5_p0.05<-as.data.frame(res0_0.5_p0.05)
```

```{r}
# Include plot
# yintercept is the threshold of -log(p-value) so -log(0.05) is 1.301
library(ggplot2)
plot1 <- ggplot(data = res0_0.5_p0.05, 
            aes(x = log2FoldChange, 
                y = -log2(res0_0.5_p0.05$padj), 
                colour=expression,
                label = res0_0.5_p0.05$SYMBOL)) +
  geom_point(alpha=0.4, size=3.5) +
  scale_color_manual(values=c("blue", "grey","red"))+
  xlim(c(-4.5, 4.5)) +
  geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +
  geom_hline(yintercept = 1.301,lty=4,col="black",lwd=0.8) +
  labs(x="log2(fold change)",
       y="-log10 (adj.p-value)",
       title="Differential expression Day 0~0.5")  +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), 
        legend.position="right", 
        legend.title = element_blank())
plot1
```

Results indicate that 15529 genes with nonzero counts were assessed whereof 2.1% was upregulated and 1.4% was downregulated (p<0.1). Nevertheless, this p-value is non-significant. Therefore, I reran the piece of code for significance levels a=0.05 and a=0.01. This revealed reliable results as 1.7% of genes were upregulated and 0.77% were downregulated (p<0.05). However, I still wanted to be more stringent in my analysis which eventually led to 1.1% of genes upregulating and 0.3% of genes downregulating (p<0.01).

## Differential expression analysis - Comparison timepoint 0.5 and 1 ##
```{r}
# call function
diffex_0.5_1<-DESeq(dds0.5_1)
# build results table
res0.5_1 <- results(diffex_0.5_1)
res0.5_1
summary(res0.5_1)
# Look at results when p-value is 0.05
res0.5_1_p0.05<- results(diffex_0.5_1, alpha = 0.05)
summary(res0.5_1_p0.05)
# Look at results when p-value is 0.01
res0.5_1_p0.01<- results(diffex_0.5_1, alpha = 0.01)
summary(res0.5_1_p0.01)
```

```{r}
# Determine expression labels; up down 
res0.5_1_p0.05$expression = ifelse(res0.5_1_p0.05$padj < 0.05 & abs(res0.5_1_p0.05$log2FoldChange) >= 1, 
                     ifelse(res0.5_1_p0.05$log2FoldChange> 1 ,'Up','Down'),
                     'Stable')
res0.5_1_p0.05<-as.data.frame(res0.5_1_p0.05)

# Include plot
# yintercept is the threshold of -log(p-value) so -log(0.05) is 1.301
library(ggplot2)
plot2 <- ggplot(data = res0.5_1_p0.05, 
            aes(x = log2FoldChange, 
                y = -log2(res0.5_1_p0.05$padj), 
                colour=expression,
                label = res0.5_1_p0.05$SYMBOL)) +
  geom_point(alpha=0.4, size=3.5) +
  scale_color_manual(values=c("blue", "grey","red"))+
  xlim(c(-4.5, 4.5)) +
  geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +
  geom_hline(yintercept = 1.301,lty=4,col="black",lwd=0.8) +
  labs(x="log2(fold change)",
       y="-log10 (adj.p-value)",
       title="Differential expression Day ")  +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), 
        legend.position="right", 
        legend.title = element_blank())
plot2
```
Results indicate that 15296 genes with nonzero counts were assessed whereof 0.42% was upregulated and 0.6% was downregulated (p<0.1). Nevertheless, this p-value is non-significant. Therefore, I reran the piece of code for significance levels a=0.05 and a=0.01. This revealed reliable results as 0.24% of genes were upregulated and 0.35% were downregulated (p<0.05). However, I still wanted to be more stringent in my analysis which eventually led to 0.092% of genes upregulating and 0.18% of genes downregulating (p<0.01).

## Differential expression analysis - Comparison timepoint 1 and 1.5 ##
```{r}
# call function
diffex_1_1.5<-DESeq(dds1_1.5)
# build results table
res1_1.5 <- results(diffex_1_1.5)
res1_1.5
summary(res1_1.5)
# Look at results when p-value is 0.05
res1_1.5_p0.05<- results(diffex_1_1.5, alpha = 0.05)
summary(res1_1.5_p0.05)
# Look at results when p-value is 0.01
res1_1.5_p0.01<- results(diffex_1_1.5, alpha = 0.01)
summary(res1_1.5_p0.01)
```

```{r}
# Determine expression labels; up down 
res1_1.5_p0.05$expression = ifelse(res1_1.5_p0.05$padj < 0.05 & abs(res1_1.5_p0.05$log2FoldChange) >= 1, 
                     ifelse(res1_1.5_p0.05$log2FoldChange> 1 ,'Up','Down'),
                     'Stable')
res1_1.5_p0.05<-as.data.frame(res1_1.5_p0.05)

# Include plot
# yintercept is the threshold of -log(p-value) so -log(0.05) is 1.301
library(ggplot2)
plot3 <- ggplot(data = res1_1.5_p0.05, 
            aes(x = log2FoldChange, 
                y = -log2(res1_1.5_p0.05$padj), 
                colour=expression,
                label = res1_1.5_p0.05$SYMBOL)) +
  geom_point(alpha=0.4, size=3.5) +
  scale_color_manual(values=c("blue", "grey","red"))+
  xlim(c(-4.5, 4.5)) +
  geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +
  geom_hline(yintercept = 1.301,lty=4,col="black",lwd=0.8) +
  labs(x="log2(fold change)",
       y="-log10 (adj.p-value)",
       title="Differential expression Day 1~1.5")  +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), 
        legend.position="right", 
        legend.title = element_blank())
plot3
```
Results indicate that 19593 genes with nonzero counts were assessed whereof 0.97% was upregulated and 0.86% was downregulated (p<0.1). Nevertheless, this p-value is non-significant. Therefore, I reran the piece of code for significance levels a=0.05 and a=0.01. This revealed reliable results as 0.83% of genes were upregulated and 0.56% were downregulated (p<0.05). However, I still wanted to be more stringent in my analysis which eventually led to 0.42% of genes upregulating and 0.23% of genes downregulating (p<0.01).


## Differential expression analysis - Comparison timepoint 1.5 and 2 ##

```{r}
# call function
diffex_1.5_2<-DESeq(dds1.5_2)
# build results table
res1.5_2 <- results(diffex_1.5_2)
res1.5_2
summary(res1.5_2)
# Look at results when p-value is 0.05
res1.5_2_p0.05<- results(diffex_1.5_2, alpha = 0.05)
summary(res1.5_2_p0.05)
# Look at results when p-value is 0.01
res1.5_2_p0.01<- results(diffex_1.5_2, alpha = 0.01)
summary(res1.5_2_p0.01)
```

```{r}
res1.5_2_p0.05$expression = ifelse(res1.5_2_p0.05$padj < 0.05 & abs(res1.5_2_p0.05$log2FoldChange) >= 1, 
                     ifelse(res1.5_2_p0.05$log2FoldChange> 1 ,'Up','Down'),
                     'Stable')
res1.5_2_p0.05<-as.data.frame(res1.5_2_p0.05)

# Include plot
# yintercept is the threshold of -log(p-value) so -log(0.05) is 1.301
library(ggplot2)
plot4 <- ggplot(data = res1.5_2_p0.05, 
            aes(x = log2FoldChange, 
                y = -log2(res1.5_2_p0.05$padj), 
                colour=expression,
                label = res1.5_2_p0.05$SYMBOL)) +
  geom_point(alpha=0.4, size=3.5) +
  scale_color_manual(values=c("blue", "grey","red"))+
  xlim(c(-4.5, 4.5)) +
  geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +
  geom_hline(yintercept = 1.301,lty=4,col="black",lwd=0.8) +
  labs(x="log2(fold change)",
       y="-log10 (adj.p-value)",
       title="Differential expression Day 1.5~2")  +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), 
        legend.position="right", 
        legend.title = element_blank())
plot4
```
Results indicate that 19593 genes with nonzero counts were assessed whereof 0.97% was upregulated and 0.86% was downregulated (p<0.1). Nevertheless, this p-value is non-significant. Therefore, I reran the piece of code for significance levels a=0.05 and a=0.01. This revealed reliable results as 0.93% of genes were upregulated and 1.6% were downregulated (p<0.05). However, I still wanted to be more stringent in my analysis which eventually led to 0.56% of genes upregulating and 1% of genes downregulating (p<0.01).

## Differential expression analysis - Comparison timepoint 2 and 2.5 ##
```{r}
# call function
diffex_2_2.5<-DESeq(dds2_2.5)
# build results table
res2_2.5 <- results(diffex_2_2.5)
res2_2.5
summary(res2_2.5)
# Look at results when p-value is 0.05
res2_2.5_p0.05<- results(diffex_2_2.5, alpha = 0.05)
summary(res2_2.5_p0.05)
# Look at results when p-value is 0.01
res2_2.5_p0.01<- results(diffex_2_2.5, alpha = 0.01)
summary(res2_2.5_p0.01)
```

```{r}
res2_2.5_p0.05$expression = ifelse(res2_2.5_p0.05$padj < 0.05 & abs(res2_2.5_p0.05$log2FoldChange) >= 1, 
                     ifelse(res2_2.5_p0.05$log2FoldChange> 1 ,'Up','Down'),
                     'Stable')
res2_2.5_p0.05<-as.data.frame(res2_2.5_p0.05)

# Include plot
# yintercept is the threshold of -log(p-value) so -log(0.05) is 1.301
library(ggplot2)
plot4 <- ggplot(data = res2_2.5_p0.05, 
            aes(x = log2FoldChange, 
                y = -log2(res2_2.5_p0.05$padj), 
                colour=expression,
                label = res2_2.5_p0.05$SYMBOL)) +
  geom_point(alpha=0.4, size=3.5) +
  scale_color_manual(values=c("blue", "grey","red"))+
  xlim(c(-4.5, 4.5)) +
  geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +
  geom_hline(yintercept = 1.301,lty=4,col="black",lwd=0.8) +
  labs(x="log2(fold change)",
       y="-log10 (adj.p-value)",
       title="Differential expression Day 2~2.5")  +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), 
        legend.position="right", 
        legend.title = element_blank())
plot4
```
Results indicate that 15481 genes with nonzero counts were assessed whereof 0.75% was upregulated and 1.8% was downregulated (p<0.1). Nevertheless, this p-value is non-significant. Therefore, I reran the piece of code for significance levels a=0.05 and a=0.01. This revealed reliable results as 0.48% of genes were upregulated and 1.5% were downregulated (p<0.05). However, I still wanted to be more stringent in my analysis which eventually led to 0.25% of genes upregulating and 1.1% of genes downregulating (p<0.01).

## Differential expression analysis - Comparison timepoint 2.5 and 3 ##
```{r}
# call function
diffex_2.5_3<-DESeq(dds2.5_3)
# build results table
res2.5_3 <- results(diffex_2.5_3)
res2.5_3
summary(res2.5_3)
# Look at results when p-value is 0.05
res2.5_3_p0.05<- results(diffex_2.5_3, alpha = 0.05)
summary(res2.5_3_p0.05)
# Look at results when p-value is 0.01
res2.5_3_p0.01<- results(diffex_2.5_3, alpha = 0.01)
summary(res2.5_3_p0.01)
```

```{r}
res2.5_3_p0.05$expression = ifelse(res2.5_3_p0.05$padj < 0.05 & abs(res2.5_3_p0.05$log2FoldChange) >= 1, 
                     ifelse(res2.5_3_p0.05$log2FoldChange> 1 ,'Up','Down'),
                     'Stable')
res2.5_3_p0.05<-as.data.frame(res2.5_3_p0.05)

# Include plot
# yintercept is the threshold of -log(p-value) so -log(0.05) is 1.301
library(ggplot2)
plot4 <- ggplot(data = res2.5_3_p0.05, 
            aes(x = log2FoldChange, 
                y = -log2(res2.5_3_p0.05$padj), 
                colour=expression,
                label = res2.5_3_p0.05$SYMBOL)) +
  geom_point(alpha=0.4, size=3.5) +
  scale_color_manual(values=c("blue", "grey","red"))+
  xlim(c(-4.5, 4.5)) +
  geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +
  geom_hline(yintercept = 1.301,lty=4,col="black",lwd=0.8) +
  labs(x="log2(fold change)",
       y="-log10 (adj.p-value)",
       title="Differential expression Day 2.5~3")  +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), 
        legend.position="right", 
        legend.title = element_blank())
plot4
```
Results indicate that 15630 genes with nonzero counts were assessed whereof 0.16% was upregulated and 0.34% was downregulated (p<0.1). Nevertheless, this p-value is non-significant. Therefore, I reran the piece of code for significance levels a=0.05 and a=0.01. This revealed reliable results as 0.1% of genes were upregulated and 0.26% were downregulated (p<0.05). However, I still wanted to be more stringent in my analysis which eventually led to 0.058% of genes upregulating and 0.17% of genes downregulating (p<0.01).    


## Differential expression analysis - Comparison timepoint 3 and 4.5 ##
```{r}
# call function
diffex_3_4.5<-DESeq(dds3_4.5)
# build results table
res3_4.5 <- results(diffex_3_4.5)
res3_4.5
summary(res3_4.5)
# Look at results when p-value is 0.05
res3_4.5_p0.05<- results(diffex_3_4.5, alpha = 0.05)
summary(res3_4.5_p0.05)
# Look at results when p-value is 0.01
res3_4.5_p0.01<- results(diffex_3_4.5, alpha = 0.01)
summary(res3_4.5_p0.01)
```

```{r}
res3_4.5_p0.05$expression = ifelse(res3_4.5_p0.05$padj < 0.05 & abs(res3_4.5_p0.05$log2FoldChange) >= 1, 
                     ifelse(res3_4.5_p0.05$log2FoldChange> 1 ,'Up','Down'),
                     'Stable')
res3_4.5_p0.05<-as.data.frame(res3_4.5_p0.05)

# Include plot
# yintercept is the threshold of -log(p-value) so -log(0.05) is 1.301
library(ggplot2)
plot5 <- ggplot(data = res3_4.5_p0.05, 
            aes(x = log2FoldChange, 
                y = -log2(res3_4.5_p0.05$padj), 
                colour=expression,
                label = res3_4.5_p0.05$SYMBOL)) +
  geom_point(alpha=0.4, size=3.5) +
  scale_color_manual(values=c("blue", "grey","red"))+
  xlim(c(-4.5, 4.5)) +
  geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +
  geom_hline(yintercept = 1.301,lty=4,col="black",lwd=0.8) +
  labs(x="log2(fold change)",
       y="-log10 (adj.p-value)",
       title="Differential expression Day 3~4.5")  +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), 
        legend.position="right", 
        legend.title = element_blank())
plot5
```
## Differential expression analysis - Comparison timepoint 4.5 and 5 ##
```{r}
# call function
diffex_4.5_5<-DESeq(dds4.5_5)
# build results table
res4.5_5 <- results(diffex_4.5_5)
res4.5_5
summary(res4.5_5)
# Look at results when p-value is 0.05
res4.5_5_p0.05<- results(diffex_4.5_5, alpha = 0.05)
summary(res4.5_5_p0.05)
# Look at results when p-value is 0.01
res4.5_5_p0.01<- results(diffex_4.5_5, alpha = 0.01)
summary(res4.5_5_p0.01)
```
```{r}
res4.5_5_p0.05$expression = ifelse(res4.5_5_p0.05$padj < 0.05 & abs(res4.5_5_p0.05$log2FoldChange) >= 1, 
                     ifelse(res4.5_5_p0.05$log2FoldChange> 1 ,'Up','Down'),
                     'Stable')
res4.5_5_p0.05<-as.data.frame(res4.5_5_p0.05)

# Include plot
# yintercept is the threshold of -log(p-value) so -log(0.05) is 1.301
library(ggplot2)
plot8 <- ggplot(data = res4.5_5_p0.05, 
            aes(x = log2FoldChange, 
                y = -log2(res4.5_5_p0.05$padj), 
                colour=expression,
                label = res3_4.5_p0.05$SYMBOL)) +
  geom_point(alpha=0.4, size=3.5) +
  scale_color_manual(values=c("blue", "grey","red"))+
  xlim(c(-4.5, 4.5)) +
  geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +
  geom_hline(yintercept = 1.301,lty=4,col="black",lwd=0.8) +
  labs(x="log2(fold change)",
       y="-log10 (adj.p-value)",
       title="Differential expression Day 4.5~5")  +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), 
        legend.position="right", 
        legend.title = element_blank())
plot8
```
## Differential expression analysis - Comparison timepoint 5 and 6 ##
```{r}
# call function
diffex_5_6<-DESeq(dds5_6)
# build results table
res5_6 <- results(diffex_5_6)
res5_6
summary(res5_6)
# Look at results when p-value is 0.05
res5_6_p0.05<- results(diffex_5_6, alpha = 0.05)
summary(res5_6_p0.05)
# Look at results when p-value is 0.01
res5_6_p0.01<- results(diffex_5_6, alpha = 0.01)
summary(res5_6_p0.01)
```
```{r}
res5_6_p0.05$expression = ifelse(res5_6_p0.05$padj < 0.05 & abs(res5_6_p0.05$log2FoldChange) >= 1, 
                     ifelse(res5_6_p0.05$log2FoldChange> 1 ,'Up','Down'),
                     'Stable')
res5_6_p0.05<-as.data.frame(res5_6_p0.05)

# Include plot
# yintercept is the threshold of -log(p-value) so -log(0.05) is 1.301
library(ggplot2)
plot9 <- ggplot(data = res5_6_p0.05, 
            aes(x = log2FoldChange, 
                y = -log2(res5_6_p0.05$padj), 
                colour=expression,
                label = res3_4.5_p0.05$SYMBOL)) +
  geom_point(alpha=0.4, size=3.5) +
  scale_color_manual(values=c("blue", "grey","red"))+
  xlim(c(-4.5, 4.5)) +
  geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +
  geom_hline(yintercept = 1.301,lty=4,col="black",lwd=0.8) +
  labs(x="log2(fold change)",
       y="-log10 (adj.p-value)",
       title="Differential expression Day 5~6")  +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), 
        legend.position="right", 
        legend.title = element_blank())
plot9
```

```{r}
# call function
diffex_6_0<-DESeq(dds6_0)
# build results table
res6_0 <- results(diffex_6_0)
res6_0
summary(res6_0)
# Look at results when p-value is 0.05
res6_0_p0.05<- results(diffex_6_0, alpha = 0.05)
summary(res6_0_p0.05)
# Look at results when p-value is 0.01
res6_0_p0.01<- results(diffex_6_0, alpha = 0.01)
summary(res6_0_p0.01)

res6_0_p0.05$expression = ifelse(res6_0_p0.05$padj < 0.05 & abs(res6_0_p0.05$log2FoldChange) >= 1, 
                     ifelse(res6_0_p0.05$log2FoldChange> 1 ,'Up','Down'),
                     'Stable')
res6_0_p0.05<-as.data.frame(res6_0_p0.05)
```

```{r}
# convert vst counts to df
gene_expr_0_0.5<-as.data.frame(gene_expr_0_0.5)
gene_expr_0.5_1<-as.data.frame(gene_expr_0.5_1)
gene_expr_1_1.5<-as.data.frame(gene_expr1_1.5)
gene_expr_1.5_2<-as.data.frame(gene_expr_1.5_2)
gene_expr_2_2.5<-as.data.frame(gene_expr_2_2.5)
gene_expr_2.5_3<-as.data.frame(gene_expr_2.5_3)
gene_expr_3_4.5<-as.data.frame(gene_expr_3_4.5)
gene_expr_4.5_5<-as.data.frame(gene_expr_4.5_5)
gene_expr_5_6<-as.data.frame(gene_expr_5_6)
gene_expr_6_0<-as.data.frame(gene_expr_6_0)

# convert rownames back to columns in vst count results
gene_expr_0_0.5<-rownames_to_column(gene_expr_0_0.5)
gene_expr_0.5_1<-rownames_to_column(gene_expr_0.5_1)
gene_expr_1_1.5<-rownames_to_column(gene_expr_1_1.5)
gene_expr_1.5_2<-rownames_to_column(gene_expr_1.5_2)
gene_expr_2_2.5<-rownames_to_column(gene_expr_2_2.5)
gene_expr_2.5_3<-rownames_to_column(gene_expr_2.5_3)
gene_expr_3_4.5<-rownames_to_column(gene_expr_3_4.5)
gene_expr_4.5_5<-rownames_to_column(gene_expr_4.5_5)
gene_expr_5_6<-rownames_to_column(gene_expr_5_6)
gene_expr_6_0<-rownames_to_column(gene_expr_6_0)

write_xlsx(gene_expr_0_0.5, "gene_expr_0_0.5.xlsx")
write_xlsx(as.data.frame(gene_expr_0.5_1), "gene_expr_0.5_1.xlsx")
write_xlsx(as.data.frame(gene_expr1_1.5), "gene_expr_1_1.5.xlsx")
write_xlsx(as.data.frame(gene_expr_1.5_2), "gene_expr_1.5_2.xlsx")
write_xlsx(as.data.frame(gene_expr_2_2.5), "gene_expr_2_2.5.xlsx")
write_xlsx(as.data.frame(gene_expr_2.5_3), "gene_expr_2.5_3.xlsx")
write_xlsx(as.data.frame(gene_expr_3_4.5), "gene_expr_3_4.5.xlsx")
write_xlsx(as.data.frame(gene_expr_4.5_5), "gene_expr_4.5_5.xlsx")
write_xlsx(as.data.frame(gene_expr_5_6), "gene_expr_5_6.xlsx")
write_xlsx(as.data.frame(gene_expr_6_0), "gene_expr_6_0.xlsx")


# convert rownames back to columns in deseq results
res0_0.5_p0.05<-rownames_to_column(res0_0.5_p0.05)
res0.5_1_p0.05<-rownames_to_column(res0.5_1_p0.05)
res1_1.5_p0.05<-rownames_to_column(res1_1.5_p0.05)
res1.5_2_p0.05<-rownames_to_column(res1.5_2_p0.05)
res2_2.5_p0.05<-rownames_to_column(res2_2.5_p0.05)
res2.5_3_p0.05<-rownames_to_column(res2.5_3_p0.05)
res3_4.5_p0.05<-rownames_to_column(res3_4.5_p0.05)
res4.5_5_p0.05<-rownames_to_column(res4.5_5_p0.05)
res5_6_p0.05<-rownames_to_column(res5_6_p0.05)
res6_0_p0.05<-rownames_to_column(res6_0_p0.05)

# export deseq results
write_xlsx(res0_0.5_p0.05, "Deseq_Results_0_0.5_p0.05.xlsx")
write_xlsx(res0.5_1_p0.05, "Deseq_Results_0.5_1_p0.05.xlsx")
write_xlsx(res1_1.5_p0.05, "Deseq_Results_1_1.5_p0.05.xlsx")
write_xlsx(res1.5_2_p0.05, "Deseq_Results_1.5_2_p0.05.xlsx")
write_xlsx(res2_2.5_p0.05, "Deseq_Results_2_2.5_p0.05.xlsx")
write_xlsx(res2.5_3_p0.05, "Deseq_Results_2.5_3_p0.05.xlsx")
write_xlsx(res3_4.5_p0.05, "Deseq_Results_3_4.5_p0.05.xlsx")
write_xlsx(res4.5_5_p0.05, "Deseq_Results_4.5_5_p0.05.xlsx")
write_xlsx(res5_6_p0.05, "Deseq_Results_5_6_p0.05.xlsx")
write_xlsx(res6_0_p0.05, "Deseq_Results_6_0_p0.05.xlsx")

```

